pkg/nsx/services/vpc/vpc.go

```diff
@@ -1318,18 +1316,73 @@ func (s *VPCService) GetAllVPCsFromNSX() map[string]model.Vpc {
 // GetNamespacesWithPreCreatedVPCs returns a map of the Namespaces which use the pre-created VPCs. The
 // key of the map is the Namespace name, and the value is the pre-created VPC path used in the NetworkInfo
 // within this Namespace.
-func (s *VPCService) GetNamespacesWithPreCreatedVPCs() map[string]string {
+func (s *VPCService) GetNamespacesWithPreCreatedVPCs() (map[string]string, error) {
 	nsVpcMap := make(map[string]string)
-	for ncName, cfg := range s.VPCNetworkConfigStore.VPCNetworkConfigMap {
-		if IsPreCreatedVPC(cfg) {
-			for _, ns := range s.GetNamespacesByNetworkconfigName(ncName) {
-				nsVpcMap[ns] = cfg.VPCPath
+
+	vpcNetworkConfigList := &v1alpha1.VPCNetworkConfigurationList{}
+	err := s.Client.List(context.Background(), vpcNetworkConfigList)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cfgCR := range vpcNetworkConfigList.Items {
+		if IsPreCreatedVPC(common.VPCNetworkConfigInfo{VPCPath: cfgCR.Spec.VPC}) {
+			nsList, err := s.GetNamespacesByNetworkconfigName(cfgCR.Name)
+			if err != nil {
+				return nsVpcMap, err
+			}
+			for _, ns := range nsList {
+				nsVpcMap[ns] = cfgCR.Spec.VPC
 			}
 		}
 	}
-	return nsVpcMap
+	return nsVpcMap, nil
 }
 
 func IsPreCreatedVPC(nc common.VPCNetworkConfigInfo) bool {
 	return nc.VPCPath != ""
 }
+
+func buildNetworkConfigInfo(vpcConfigCR *v1alpha1.VPCNetworkConfiguration) (*common.VPCNetworkConfigInfo, error) {
+	org, project, err := nsxProjectPathToId(vpcConfigCR.Spec.NSXProject)
+	if err != nil {
+		log.Error(err, "Failed to parse NSX project in NetworkConfig", "Project Path", vpcConfigCR.Spec.NSXProject)
+		return nil, err
+	}
+
+	ninfo := &common.VPCNetworkConfigInfo{
+		IsDefault:              isDefaultNetworkConfigCR(vpcConfigCR),
+		Org:                    org,
+		Name:                   vpcConfigCR.Name,
+		VPCConnectivityProfile: vpcConfigCR.Spec.VPCConnectivityProfile,
+		NSXProject:             project,
+		PrivateIPs:             vpcConfigCR.Spec.PrivateIPs,
+		DefaultSubnetSize:      vpcConfigCR.Spec.DefaultSubnetSize,
+		VPCPath:                vpcConfigCR.Spec.VPC,
+	}
+	return ninfo, nil
+}
+
+// parse org id and project id from nsxProject path
+// example /orgs/default/projects/nsx_operator_e2e_test
+func nsxProjectPathToId(path string) (string, string, error) {
+	parts := strings.Split(path, "/")
+	if len(parts) < 4 {
+		return "", "", errors.New("invalid NSX project path")
+	}
+	return parts[2], parts[len(parts)-1], nil
+}
+
+func isDefaultNetworkConfigCR(vpcConfigCR *v1alpha1.VPCNetworkConfiguration) bool {
+	annos := vpcConfigCR.GetAnnotations()
+	val, exist := annos[common.AnnotationDefaultNetworkConfig]
```

Comment:
the same function isDefaultNetworkConfigCR/nsxProjectPathToId in the vpcnetworkconfig_handler.go, could you  reuse them?